src/components/controllers/repl/examples/todo-lists/todo-list-signals.txt

```diff
@@ -0,0 +1,58 @@
+import { render } from 'preact';
+import { signal, computed } from '@preact/signals';
+
+const todos = signal([
+	{ text: 'Write my first post', completed: true },
+	{ text: 'Buy new groceries', completed: false },
+	{ text: 'Walk the dog', completed: false }
+]);
+
+const completedCount = computed(
+	() => todos.value.filter(todo => todo.completed).length
+);
+
+const newItem = signal('');
+
+function addTodo(e) {
+	e.preventDefault();
+
+	todos.value = [...todos.value, { text: newItem.value, completed: false }];
+	newItem.value = ''; // Reset input value on add
+}
+
+function removeTodo(index) {
+	todos.value.splice(index, 1);
+	todos.value = [...todos.value];
+}
+
+function TodoList() {
+	const onInput = event => (newItem.value = event.target.value);
+
+	return (
+		<form onSubmit={addTodo}>
+			<input type="text" value={newItem.value} onInput={onInput} />
+			<button onClick={addTodo}>Add</button>
+			<ul>
+				{todos.value.map((todo, index) => (
+					<li>
+						<label>
+							<input
+								type="checkbox"
+								checked={todo.completed}
+								onInput={() => {
+									todo.completed = !todo.completed;
+									todos.value = [...todos.value];
+								}}
+							/>
+							{todo.completed ? <s>{todo.text}</s> : todo.text}
+						</label>{' '}
+						<button onClick={() => removeTodo(index)}>❌</button>
+					</li>
+				))}
+			</ul>
+			<p>Completed count: {completedCount.value}</p>
+		</form>
+	);
+}
```

Comment:
The updated example doesn't work. Try removing a todo.

Reply:
Seems to work just fine for me? What issue are you seeing?

Reply:
The remove button doesn't actually remove the todo item, it only empties the text. The previous example works just fine.

![preactjs](https://github.com/user-attachments/assets/e19d89aa-7c30-4c61-99f8-ee374b7105e4)

Reply:
Strange, I can't reproduce. Try refreshing whilst clearing the cache, you might have something left over? Else, I'll look into it when I can.

https://github.com/user-attachments/assets/0117094d-9c3e-4351-beb7-e4d1077b5b68

Reply:
Refreshing doesn't help in my case. My browser is Chrome 128.0.6613.114 on Windows 10.

I made a longer recording, and based on my observation, it seems that clicking the remove button indeed removed the todo item, but it also triggered the form submission and added an empty todo item:

![preactjs](https://github.com/user-attachments/assets/5c1661c6-f4ba-4475-afce-b82e8b438b0a)

Reply:
Ah, yep, forgot `type="button"` as per usual.

Good catch!